deepchecks/base/dataset.py

```diff
@@ -128,12 +132,19 @@ def __init__(
                     raise DeepchecksValueError('Label must be a column vector')
 
                 # Get label name
-                if isinstance(label, (pd.Series, pd.DataFrame)) and label.name is not None:
+                if isinstance(label, pd.Series) and label.name is not None:
                     label_name = label.name
                     if label_name in self._data.columns:
                         raise DeepchecksValueError(f'Data has column with name "{label_name}", use pandas rename to'
                                                    f' change label name or remove the column from the dataframe')
-                else:
+                elif isinstance(label, pd.DataFrame):
+                    label_name = label.columns[0]
```

Comment:
Validation that supplied DataFrame has indeed only one column? (e.g. user can mix it up and send us a DataFrame with "all data incl. label")